commons/src/test/scala/dagr/commons/util/BiMapTest.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015-2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package dagr.commons.util
+
+class BiMapTest extends UnitSpec {
```

Comment:
Reading your tests makes me realize that `BiMap` has some potential problems.  Since it's a map type there should probably be some tests that assert that the association between keys and values are correctly maintained by the map.

E.g. what should the following test case produce:
```scala
val map = new BiMap[String, String]()
map.add("key1", "val1")
map.add("key2", "val1")
???
```

I believe the current implementation will yield:
```
forward:
key1 -> val1
key2 -> val1

reverse:
val1 -> key2
```

This is not intuitive behaviour!  So I think either BiMap should define that each value may have only one key (in which case `key1 -> val1` should be evicted when adding `key2 -> val1`) _or_ BiMap has `key->value` and `value->Set[Key]`.